### ISSUE TITLE
Cache ensemble model logits or logprobs

### ISSUE THREAD

**Issue by FY-Liu** [r-three/SLMensembles#37]
Current ensemble implementation holds one copy of all ensemble models on each gpu. This creates huge memory overhead and limits the speed. Ideally, ensemble model logits should be pre-computed and stored as a part of the dataset. Such logit/logprob can be accumulated with each new ensemble model for less storage space. 

Note: current implementation adds up logits before perform logsoftmax, only for the kl_div step. Such order cannot be re-ordered. The implementaiton needs to be verified.
